cmd/agentless-scanner/main.go

```diff
@@ -6,5 +6,307 @@
 // Package main implements the agentless-scanner command.
 package main
 
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/agentless/runner"
+	"github.com/DataDog/datadog-agent/pkg/agentless/types"
+
+	// DataDog agent: config stuffs
+	commonpath "github.com/DataDog/datadog-agent/cmd/agent/common/path"
+	"github.com/DataDog/datadog-agent/comp/core"
+	compconfig "github.com/DataDog/datadog-agent/comp/core/config"
+	complog "github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform/eventplatformimpl"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatformreceiver/eventplatformreceiverimpl"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/pidfile"
+	"github.com/DataDog/datadog-agent/pkg/security/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	ddogstatsd "github.com/DataDog/datadog-go/v5/statsd"
+
+	_ "modernc.org/sqlite" // sqlite driver, used by github.com/knqyf263/go-rpmdb
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultWorkersCount = 15
+	defaultScannersMax  = 8 // max number of child-process scanners spawned by a worker in parallel
+)
+
+var statsd *ddogstatsd.Client
+var eventForwarder eventplatform.Component
+var config compconfig.Component
```

Comment:
We should avoid using globals as much as possible. I spart of one the OKR for the ASC team. 

Could we propagate the component down to the commands and use them, rather than relaying on globals?

Reply:
Done.